spatialmath/mesh.go

```diff
@@ -18,15 +21,62 @@ type Mesh struct {
 	pose      Pose
 	triangles []*Triangle
 	label     string
+
+	// information used for encoding to protobuf
+	fileType meshType
+	rawBytes []byte
+}
+
+// The set of supported mesh file types.
+type meshType string
+
+const plyType = meshType("ply")
+
+func newMeshFromProto(pose Pose, m *commonpb.Mesh, label string) (*Mesh, error) {
+	switch m.FileType {
+	case string(plyType):
+		return NewMeshFromPLY(pose, m.Mesh, label)
+	default:
+		return nil, fmt.Errorf("unsupported Mesh type: %s", m.FileType)
+	}
 }
 
-// NewMesh creates a mesh from the given triangles and pose.
-func NewMesh(pose Pose, triangles []*Triangle, label string) *Mesh {
```

Comment:
I have found this useful for testing things where I hand-jam triangles such as for the coverage algos